aws/resource_aws_msk_scram_secret_test.go

```diff
@@ -0,0 +1,359 @@
+	expectedDelete := []string{"AmazonMSK_test_aws_example_test_tf-test-4-I50lRm"}
+
+	updated, deleted := filterMskSecretsForDeletion(newSecrets, existingSecrets)
+
+	if !reflect.DeepEqual(expectedUpdate, aws.StringValueSlice(updated)) {
+		t.Fatalf("Expected secret list to be %v, got %v", expectedUpdate, aws.StringValueSlice(updated))
+	}
+
+	if !reflect.DeepEqual(expectedDelete, aws.StringValueSlice(deleted)) {
+		t.Fatalf("Expected secret list to be %v, got %v", expectedDelete, aws.StringValueSlice(deleted))
+	}
+}
+
+func testAccCheckAwsSaslScramSecretsDontExist(resourceName string, count int) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		if len(rs.Primary.Attributes["secret_arn_list"]) > count {
+			return fmt.Errorf("Too many secrets for %v", rs.Primary.Attributes["secret_arn_list"])
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAwsSaslScramSecretsDestruction(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		_, ok := s.RootModule().Resources[resourceName]
+		if ok {
+			return fmt.Errorf("Should not have found: %s", resourceName)
+		}
+
+		return nil
+	}
+}
+
+func testAccMskSaslScramSecret_basic(rName string) string {
+	return testAccMskClusterBaseConfig() + fmt.Sprintf(`
+resource "aws_msk_cluster" "test" {
+  cluster_name           = %[1]q
+  kafka_version          = "2.5.1"
+  number_of_broker_nodes = 3
+
+  broker_node_group_info {
+    client_subnets  = [aws_subnet.example_subnet_az1.id, aws_subnet.example_subnet_az2.id, aws_subnet.example_subnet_az3.id]
+    ebs_volume_size = 10
+    instance_type   = "kafka.t3.small"
+    security_groups = [aws_security_group.example_sg.id]
+  }
+
+  client_authentication {
+    sasl {
+      scram = true
+    }
+  }
+}
+
+resource "aws_kms_key" "test" {
+  description = "%s-kms-key-msk"
+}
+
+resource "aws_secretsmanager_secret" "test_basic" {
+  name       = "AmazonMSK_test_%s_1_basic"
+  kms_key_id = aws_kms_key.test.key_id
+}
+
+resource "aws_secretsmanager_secret_version" "test" {
+  secret_id     = aws_secretsmanager_secret.test_basic.id
+  secret_string = jsonencode({ username = "%s", password = "foobar" })
+}
+
+resource "aws_msk_sasl_scram_secret" "test" {
+  cluster_arn     = aws_msk_cluster.test.arn
+  secret_arn_list = [aws_secretsmanager_secret.test_basic.arn]
+}
+
+resource "aws_secretsmanager_secret_policy" "test" {
+  secret_arn = aws_secretsmanager_secret.test_basic.arn
+  policy     = <<POLICY
+{
+  "Version" : "2012-10-17",
+  "Statement" : [ {
+    "Sid": "AWSKafkaResourcePolicy",
+    "Effect" : "Allow",
+    "Principal" : {
+      "Service" : "kafka.amazonaws.com"
+    },
+    "Action" : "secretsmanager:getSecretValue",
+    "Resource" : "${aws_secretsmanager_secret.test_basic.arn}"
+  } ]
+}
+POLICY
+}
+`, rName, rName, rName, rName)
```

Comment:
Nit: we can pass just the 1 `rName` param and use the `%[1]` formatter reference to it 👍 
```suggestion
resource "aws_msk_cluster" "test" {
  cluster_name           = %[1]q
  kafka_version          = "2.5.1"
  number_of_broker_nodes = 3

  broker_node_group_info {
    client_subnets  = [aws_subnet.example_subnet_az1.id, aws_subnet.example_subnet_az2.id, aws_subnet.example_subnet_az3.id]
    ebs_volume_size = 10
    instance_type   = "kafka.t3.small"
    security_groups = [aws_security_group.example_sg.id]
  }

  client_authentication {
    sasl {
      scram = true
    }
  }
}

resource "aws_kms_key" "test" {
  description = "%[1]s-kms-key-msk"
}

resource "aws_secretsmanager_secret" "test_basic" {
  name       = "AmazonMSK_test_%[1]s_1_basic"
  kms_key_id = aws_kms_key.test.key_id
}

resource "aws_secretsmanager_secret_version" "test" {
  secret_id     = aws_secretsmanager_secret.test_basic.id
  secret_string = jsonencode({ username = "%[1]s", password = "foobar" })
}

resource "aws_msk_sasl_scram_secret" "test" {
  cluster_arn     = aws_msk_cluster.test.arn
  secret_arn_list = [aws_secretsmanager_secret.test_basic.arn]
}

resource "aws_secretsmanager_secret_policy" "test" {
  secret_arn = aws_secretsmanager_secret.test_basic.arn
  policy     = <<POLICY
{
  "Version" : "2012-10-17",
  "Statement" : [ {
    "Sid": "AWSKafkaResourcePolicy",
    "Effect" : "Allow",
    "Principal" : {
      "Service" : "kafka.amazonaws.com"
    },
    "Action" : "secretsmanager:getSecretValue",
    "Resource" : "${aws_secretsmanager_secret.test_basic.arn}"
  } ]
}
POLICY
}
`, rName)
```